pkg/utils/k8sutil.go

```diff
@@ -387,7 +381,14 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 		Labels:      labels,
```

Comment:
You are overriding the labels+metadata here, and then copying them again below, losing any labels/annotations from the funcObj template.  Delete this objectmeta override (but first add a unittest that demonstrates this bug - because preserving annotations from the funcObj template is actually a use case we care about).